sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/aio/_azureappconfigurationproviderasync.py

```diff
@@ -403,12 +411,18 @@ async def refresh(self, **kwargs) -> None:
             if not success:
                 self._refresh_timer.backoff()
             elif need_refresh and self._on_refresh_success:
-                self._on_refresh_success()
+                await self._on_refresh_success()
```

Comment:
Do we need await for self._on_refresh_error(e) &  self._refresh_lock.release()?

Reply:
Yes for `_on_refresh_error(e)` that is a customer provided method that should be async. I don't think we are using an async lock so we don't need to await that release.

Reply:
Does sync lock work correctly in an async method?

Reply:
@annatisch, help me setup this original. I think it does work correctly.